stackit/internal/services/iaas/network/resource.go

```diff
@@ -450,59 +601,129 @@ func toCreatePayload(ctx context.Context, model *Model) (*iaas.CreateNetworkPayl
 	if model == nil {
 		return nil, fmt.Errorf("nil model")
 	}
+	addressFamily := &iaas.CreateNetworkAddressFamily{}
 
-	modelNameservers := []string{}
-	for _, ns := range model.Nameservers.Elements() {
-		nameserverString, ok := ns.(types.String)
+	modelIPv6Nameservers := []string{}
+	for _, ipv6ns := range model.IPv6Nameservers.Elements() {
+		ipv6NameserverString, ok := ipv6ns.(types.String)
 		if !ok {
 			return nil, fmt.Errorf("type assertion failed")
 		}
-		modelNameservers = append(modelNameservers, nameserverString.ValueString())
+		modelIPv6Nameservers = append(modelIPv6Nameservers, ipv6NameserverString.ValueString())
+	}
+
+	if !(model.IPv6Prefix.IsNull() || model.IPv6PrefixLength.IsNull() || model.IPv6Nameservers.IsNull()) {
+		addressFamily.Ipv6 = &iaas.CreateNetworkIPv6Body{
+			Nameservers:  &modelIPv6Nameservers,
+			Gateway:      iaas.NewNullableString(conversion.StringValueToPointer(model.IPv6Gateway)),
+			Prefix:       conversion.StringValueToPointer(model.IPv6Prefix),
+			PrefixLength: conversion.Int64ValueToPointer(model.IPv6PrefixLength),
+		}
+	}
+
+	modelIPv4Nameservers := []string{}
+	var modelIPv4List []attr.Value
+
+	if !(model.IPv4Nameservers.IsNull() || model.IPv4Nameservers.IsUnknown()) {
+		modelIPv4List = model.IPv4Nameservers.Elements()
+	} else {
+		modelIPv4List = model.Nameservers.Elements()
+	}
+
+	for _, ipv4ns := range modelIPv4List {
+		ipv4NameserverString, ok := ipv4ns.(types.String)
+		if !ok {
+			return nil, fmt.Errorf("type assertion failed")
+		}
+		modelIPv4Nameservers = append(modelIPv4Nameservers, ipv4NameserverString.ValueString())
+	}
+
+	if !model.IPv4Prefix.IsNull() || !model.IPv4PrefixLength.IsNull() || (!model.IPv4Nameservers.IsNull() && len(model.IPv4Nameservers.Elements()) > 0) || (!model.Nameservers.IsNull() && len(model.Nameservers.Elements()) > 0) {
```

Comment:
```suggestion
	if !model.IPv4Prefix.IsNull() || !model.IPv4PrefixLength.IsNull() || !model.IPv4Nameservers.IsNull() || !model.Nameservers.IsNull() {
```

Reply:
I think this is not needed, otherwise we would not be able to set empty nameservers